specification/saas/resource-manager/Microsoft.SaaS/preview/2018-03-01-beta/saas.json

```diff
@@ -0,0 +1,866 @@
+        }
+      }
+    },
+    "SaasAppPlan": {
+      "description": "Saas resource plan.",
+      "type": "object",
+      "properties": {
+        "publisher": {
+          "description": "the publisher id.",
+          "type": "string"
+        },
+        "product": {
+          "description": "the offer id.",
+          "type": "string"
+        },
+        "name": {
+          "description": "the plan id.",
+          "type": "string"
+        }
+      }
+    },
+    "Tags": {
+      "description": "the resource tags.",
+      "type": "array",
+      "items": {
+        "additionalProperties": {
+          "description": "additional properties.",
+          "type": "string"
+        }
+      }
+    },
+    "SaasAppOperation": {
+      "description": "saas app operations",
+      "type": "object",
+      "properties": {
+        "name": {
+          "description": "the operation name",
+          "type": "string"
+        },
+        "display": {
+          "description": "the operation display",
+          "$ref": "#/definitions/SaasAppOperationDisplay"
+        },
+        "origin": {
+          "description": "the operation origin",
+          "type": "string"
+        },
+        "isDataAction": {
+          "description": "whether the operation is a data action or not.",
+          "type": "boolean"
+        }
+      }
+    },
+    "SaasAppOperationDisplay": {
+      "description": "Saas app operation display",
+      "type": "object",
+      "properties": {
+        "provider": {
+          "description": "Name of the provider for display purposes",
+          "type": "string"
+        },
+        "resource": {
+          "description": "Name of the resource type for display purposes",
+          "type": "string"
+        },
+        "operation": {
+          "description": "Name of the operation for display purposes",
+          "type": "string"
+        },
+        "description": {
+          "description": "Description of the operation for display purposes",
+          "type": "string"
+        }
+      }
+    },
+    "SaasRpModernEligibility": {
+      "description": "SaasRp Modern Eligibility",
+      "type": "object",
+      "properties": {
+        "id": {
+          "description": "the azure subscription id",
+          "type": "string"
+        },
+        "isModern": {
+          "description": "whether the azure subscription has been converted to modern.",
+          "type": "boolean"
+        }
+      }
+    },
+    "deleteOptions": {
+      "description": "delete Options",
+      "type": "object",
+      "properties": {
+        "unsubscribeOnly": {
+          "type": "boolean",
+          "description": "whether it is unsubscribeOnly"
+        },
+        "reasonCode": {
+          "type": "number",
+          "description": "The reasonCode"
+        },
+        "feedback": {
+          "type": "string",
+          "description": "the feedback"
+        }
+      }
+    },
+    "ErrorResponse": {
```

Comment:
>ErrorResponse [](start = 5, length = 13)

Please take a look at [https://github.com/Azure/azure-rest-api-specs/blob/master/specification/common-types/resource-management/v2/types.json](https://github.com/Azure/azure-rest-api-specs/blob/master/specification/common-types/resource-management/v2/types.json). You should be able to reuse the error definition from that file. (Swagger allows you to reference content from another swagger file.)